server/src/main/kotlin/com/tiden/flagship/circle/CircleClient.kt

```diff
@@ -26,6 +26,100 @@ data class PaymentRequest(val idempotencyKey: String, val keyId: String, val des
+    val amount: Map<String, String>,
+    val status: String,
+    val verification: Map<String, String>? = null,
+    val cancel: Map<String, String>? = null,
+    val refunds: List<Map<String, String>>? = null,
+    val createDate: String,
+    val updateDate: String,
+    val metadata: Map<String, String>,
+    val errorCode: String? = null,
+    val riskEvaluation: Map<String, String>? = null,
+    val trackingRef: String? = null,
+    val fees: Map<String, String>? = null,
+)
+
+@Serializable
+data class PaymentResponse(
+    val data: PaymentResponseData
+)
+
+@Serializable
+data class CreateCardRequest(
+    val idempotencyKey: String,
+    val keyId: String,
+    val expMonth: Int,
+    val expYear: Int,
+    val encryptedData: String,
+    val billingDetails: Map<String, String>,
+    val metadata: Map<String, String>)
+
+@Serializable
+data class CreateCardResponseData(
+    val id: String,
+    val status: String,
+    val last4: String,
+    val billingDetails: Map<String, String>,
+    val expMonth: Int,
+    val expYear: Int,
+    val network: String,
+    val bin: String,
+    val issuerCountry: String,
+    val fundingType: String,
+    val fingerprint: String,
+    val verification: Map<String, String>,
+    val createDate: String,
+    val metadata: Map<String, String>,
+    val updateDate: String
+)
+@Serializable
+data class CreateCardResponse(
+    val data: CreateCardResponseData,
+)
+
+fun createClient() : HttpClient{
+    return HttpClient(CIO) {
+        install(JsonFeature) {
+            serializer =  KotlinxSerializer()
+        }
+    }
+}
+
+// Endpoint to store the information about the card and obtain the sourceId (Circle's internal card id) and encrypted fields
+// Encrypted card details (card number and CVV) come from the client
+// Some extra fields need to be included in the request to Circle's create card endpoint: billing details for your end user, a unique ID for the active session (sessionId) and the IP address of the end-user (ipAddress).
+// Finally, the Circle API will respond with a card containing id value (sourceId) that can be stored on our side to refer to this end-user's card in future payment requests (passed in as sourceId)
+suspend fun createCard(createCardRequest: CreateCardRequest): String {
+    val client = createClient()
+    val response: HttpResponse = client.post("https://api-sandbox.circle.com/v1/cards") {
+        val dotenv = dotenv {
+            directory = "../"
+            ignoreIfMalformed = true
+            ignoreIfMissing = true
+        }
+        val apiKey : String = dotenv["CIRCLE_API_KEY"]
+        headers {
+            append(HttpHeaders.Accept, "application/json")
+            append(HttpHeaders.Authorization, "Bearer " + apiKey)
+        }
+        contentType(ContentType.Application.Json)
+        body = createCardRequest
+        print("card creation request to Circle: " + body + "\n")
+    }
+    val responseBody: CreateCardResponse = response.receive()
```

Comment:
@popoaq here's an example of how the http `response.receive()` gets deserialized into a `CreateCardResponse` object defined above